### DISCUSSION TITLE
Is there any trick to implement cesuim script into terria?

### DISCUSSION THREAD

**Discussion by aminevsaziz** [TerriaJS/terriajs#5812] [Q&A]
Like the title suggest, is it possible to add sandcastle cesium code into terria? I'm new to both and I tried some examples on cesium castle and I want to pass that into terria.
Furthermore, is possible in terria to:
1.to customize points shape into hexagon?
2. Change line charts into bars and pies?

Thank for your help and I hope I get some help

**Comment by tephenavies** [comment]
Hi @aminevsaziz. TerriaJS doesn't have anything like Cesium Sandcastle. But it should be possible to implement anything from Sandcastle within a Terria app.

Can you confirm which major version of terriajs you're using? There are some large differences between version 7 and version 8.

To answer your other questions:
1. Customising point shapes should be possible, if it can be done in Cesium sandcastle.
2. I don't think we currently support either bar charts of pie charts, but support should be possible with modifications (check support for the type of chart you want with the underlying charting library we're using first).

To modify terriajs and build a TerriaMap against that modified terriajs follow https://docs.terria.io/guide/contributing/development-environment/

**Reply by aminevsaziz** [reply]
@steve9164  thank you for the answers. They very helpful for me and future terria users🤗.
About point 1 of customization I'm referring to points in terria not sandcastle. Is it possible to change for example color,size or even shape  of points/polygons drawn into terria using geojson/shp files.
Furthermore, my sole interest of implementing cesium code in terria is the one example that let you extrude height from geojson of osm data (file have floors  or heights column ).  But from @kring comment earlier I think (im not sure ofc)/the code can be somehow copy-paste into czml

**Reply by tephenavies** [reply]
Hi @aminevsaziz. Check out the geojson parameters "style" and "heightProperty" in this generated documentation: https://docs.terria.io/guide/connecting-to-data/catalog-type-details/geojson/. Here's an example catalog using both of those: https://gist.github.com/steve9164/d4d4faecb82e2332e0774f79cfa5b78a. You can test [here](https://beta.nationalmap.terria.io/#clean&https://gist.githubusercontent.com/steve9164/d4d4faecb82e2332e0774f79cfa5b78a/raw/d834162f584d13230ba3f819a85fb23fabc9dafb/test-geojson-styling-v8.json&share=s-d2t2uYuc6EQoncgRkjb9VqatLKy).

"marker-symbol" should support most (or all) of these icons: https://labs.mapbox.com/maki-icons/

If you want to do even more with geojson with the power of czml, check out the "czmlTemplate" parameter in the generated geojson documentation I linked above.

That is using TerriaJS version 8.
